From 2b8461a7dedc3c942eb992b9baa12bfb23bde1ce Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Mon, 11 May 2015 20:41:26 +0300
Subject: [PATCH] Contacs: check internal sdcard is non-removable

getExternalStorageState() returns status of primary sdcard
whether it's internal or external. This causes problem on
devices that have primary external sdcard. Internal sdcard
is always non-removable so also check that.

Change-Id: Iaeacf914e535ac35cfc6c58f8e5293729aebced2
---
 src/com/android/contacts/common/vcard/ExportVCardActivity.java | 3 ++-
 src/com/android/contacts/common/vcard/ImportVCardActivity.java | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/com/android/contacts/common/vcard/ExportVCardActivity.java b/src/com/android/contacts/common/vcard/ExportVCardActivity.java
index 029561d..39e63da 100644
--- a/src/com/android/contacts/common/vcard/ExportVCardActivity.java
+++ b/src/com/android/contacts/common/vcard/ExportVCardActivity.java
@@ -187,7 +187,8 @@ public class ExportVCardActivity extends Activity implements ServiceConnection,
 
     private void checkStorage() {
         boolean sdExist = MoreContactUtils.sdCardExist(this);
-        boolean inExist = Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED);
+        boolean inExist = Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)
+                && !Environment.isExternalStorageRemovable();
         if (sdExist && inExist) {
             CharSequence[] storage_list = new CharSequence[2];
             storage_list[VCardService.INTERNAL_PATH] = Environment.getExternalStorageDirectory()
diff --git a/src/com/android/contacts/common/vcard/ImportVCardActivity.java b/src/com/android/contacts/common/vcard/ImportVCardActivity.java
index d36dcaf..3bcdb5b 100644
--- a/src/com/android/contacts/common/vcard/ImportVCardActivity.java
+++ b/src/com/android/contacts/common/vcard/ImportVCardActivity.java
@@ -890,7 +890,8 @@ public class ImportVCardActivity extends Activity {
 
     private void checkStorage() {
         boolean sdExist = MoreContactUtils.sdCardExist(this);
-        boolean inExist = Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED);
+        boolean inExist = Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)
+                && !Environment.isExternalStorageRemovable();
         if (sdExist && inExist) {
             CharSequence[] storage_list = new CharSequence[2];
             storage_list[VCardService.INTERNAL_PATH] = Environment.getExternalStorageDirectory()
-- 
2.6.1

